### PR TITLE
added Bp00

### PR DESCRIPTION
**Pull request by prakharcode**

### CODE REVIEW

File: src/AstroBase.jl [hunk 1]
```diff
@@ -115,4 +115,64 @@ function tio_locator(jd1, jd2)
...
+"""
+function precession_rate_part_of_nutation(jd1, jd2)
+    PRECESSION = -sec2rad(0.29965)
+    OBLIQUITY = -sec2rad(0.02524)
+
+    t = ((jd1 - J2000) + jd2) / DAYS_PER_CENTURY
+    PRECESSION * t, OBLIQUITY * t
+end
+
+function bias_component_00()
```
Comment (giordano) [inline src/AstroBase.jl]: Does this need to be a function at all?

File: src/AstroBase.jl [hunk 2]
```diff
@@ -115,4 +115,64 @@ function tio_locator(jd1, jd2)
...
 
+"""
+    precession_rate_part_of_nutation(jd1, jd2)
+
+Returns precession corrections for a given 2 part Julian date (TT).
+
+# Example
+
+```jldoctest
+
```
Comment (giordano) [inline src/AstroBase.jl]: Are you going to add the example later? :-)

File: src/AstroBase.jl [hunk 3]
```diff
@@ -115,4 +115,64 @@ function tio_locator(jd1, jd2)
...
+# Example
+
+```jldoctest
+julia> bias_precession_matrix_00(2.4578265e6, 0.30434616919175345)
+(RotZYX(-7.07828e-8, -8.05622e-8, 3.30604e-8), [0.999991 0.00384587 0.00167106; -0.00384587 0.999993 -3.2343e-6; -0.00167106 -3.1924e-6 0.999999], [0.999991 0.00384594 0.00167098; -0.00384594 0.999993 -3.26705e-6; -0.00167098 -3.15946e-6 0.999999])
+```
+"""
+function bias_precession_matrix_00(jd1, jd2)
+    EPS0 = sec2rad(84381.448)
```
Comment (giordano) [inline src/AstroBase.jl]: Move this constant out of the function?

File: src/AstroBase.jl [hunk 4]
```diff
@@ -115,4 +115,64 @@ function tio_locator(jd1, jd2)
...
+Returns precession corrections for a given 2 part Julian date (TT).
+
+# Example
+
+```jldoctest
+
+```
+"""
+function precession_rate_part_of_nutation(jd1, jd2)
+    PRECESSION = -sec2rad(0.29965)
```
Comment (giordano) [inline src/AstroBase.jl]: Move these constants out of the function?

File: src/AstroBase.jl (hunk 1 above)
Comment (prakharcode) [reply src/AstroBase.jl]: No, we can essentially put it in constants and access the variables from there. It's our call, though the function makes it neat.

File: src/AstroBase.jl (hunk 2 above)
Comment (prakharcode) [reply src/AstroBase.jl]: yes! I'll add it later :smile:

File: src/AstroBase.jl (hunk 4 above)
Comment (prakharcode) [reply src/AstroBase.jl]: okay :)

File: src/AstroBase.jl (hunk 2 above)
Comment (helgee) [reply src/AstroBase.jl]: How about now? 😉 Then I can merge this...

File: src/AstroBase.jl [hunk 5]
```diff
@@ -485,4 +486,52 @@ function xy06(jd1, jd2)
...
+    psia77 = sec2rad((@evalpoly t 0 5038.7784 -1.07259 -0.001147))
+    oma77  = EPS0 + sec2rad(((0.05127 + (-0.007726) * t) * t) * t)
+    chia   = sec2rad((@evalpoly t 0 10.5526 -2.38064 -0.001125))
+
+    dpsipr, depspr = precession_rate_part_of_nutation(jd1, jd2)
+    psia = psia77 + dpsipr
+    oma  = oma77 + depspr
+
+    rbw = RotZYX(dra0, dpsibi*sin(EPS0), -depsbi)
+    rb = copy(rbw)
```
Comment (helgee) [inline src/AstroBase.jl]: The copy is unnecessary.

File: src/AstroBase.jl [hunk 6]
```diff
@@ -485,4 +486,52 @@ function xy06(jd1, jd2)
...
+(RotZYX(-7.07828e-8, -8.05622e-8, 3.30604e-8), [0.999991 0.00384587 0.00167106; -0.00384587 0.999993 -3.2343e-6; -0.00167106 -3.1924e-6 0.999999], [0.999991 0.00384594 0.00167098; -0.00384594 0.999993 -3.26705e-6; -0.00167098 -3.15946e-6 0.999999])
+```
+"""
+function bias_precession_matrix_00(jd1, jd2)
+    t = ((jd1 - J2000) + jd2) / DAYS_PER_CENTURY
+
+    dpsibi, depsbi, dra0 = DPBIAS, DEBIAS, DRA0
+
+    psia77 = sec2rad((@evalpoly t 0 5038.7784 -1.07259 -0.001147))
+    oma77  = EPS0 + sec2rad(((0.05127 + (-0.007726) * t) * t) * t)
```
Comment (helgee) [inline src/AstroBase.jl]: `@evalpoly`?

**Pull request merged**